### DISCUSSION TITLE
Both "ID" and "AVDID" fields in scan output are both set by avd_id metadata tag

### DISCUSSION THREAD

**Discussion by dave505** [aquasecurity/trivy#7918] [Bugs]
### Description

Setting different values for metadata tags id: and avd_id: results in only avd_id: tag being used to populate both ID and AVDID

### Desired Behavior

ID and AVDID should be different, I don't want to set AVDID for internal policies but it looks like I have to set the ID.

### Actual Behavior

ID and AVDID are the same, id: tag was ignored

### Reproduction Steps

```bash
Create a custom rego policy and set the metdata:

# METADATA
# title: some title
# description: a description
# scope: package
# schemas:
# - input: schema["cloud"]
# custom:
#   id: ID-001
#   avd_id: AVDID-001
#   recommended_action: fix this please
#   provider: aws
#   service: s3
#   severity: LOW
#   input:
#     selector:
#     - type: cloud
#       subtypes:
#         - service: s3
#           provider: aws

Results in:

        {
          "Type": "AWS",
          "ID": "AVDID-001",
          "AVDID": "AVDID-001",
          "Title": "some title",
          "Description": "a description",
          "Query": "deny",
          "Resolution": "fix this please",
          "Severity": "LOW",
```

### Target

AWS

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
debug isn't working in v0.57.0
```

### Operating System

macOS Sonoma 14.5

### Version

```bash
Version: 0.57.0
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
@nikpivkin take a look, please

**Comment by nikpivkin** [comment]
Hi @dave505 

Here's the output I got:
```json
 {
          "Type": "Terraform Security Check",
          "ID": "ID-001",
          "AVDID": "AVDID-001",
          "Title": "some title",
          "Description": "a description",
          "Message": "foo",
          "Namespace": "user.test001",
          "Query": "data.user.test001.deny",
          "Resolution": "fix this please",
          "Severity": "LOW",
          "Status": "FAIL",
```

check:
```rego
# METADATA
# title: some title
# description: a description
# scope: package
# schemas:
# - input: schema["cloud"]
# custom:
#   id: ID-001
#   avd_id: AVDID-001
#   recommended_action: fix this please
#   provider: aws
#   service: s3
#   severity: LOW
#   input:
#     selector:
#     - type: cloud
#       subtypes:
#         - service: s3
#           provider: aws

package user.test001

import rego.v1

deny contains res if {
    res := result.new("foo", {})
}
```

**Reply by dave505** [reply]
Using your example, I'm seeing

> 
      "Misconfigurations": [
        {
          "Type": "AWS",
          "ID": "AVDID-001",
          "AVDID": "AVDID-001",
          "Title": "some title",
          "Description": "a description",
          "Message": "foo",
          "Namespace": "user.test001",
          "Query": "deny",
          "Resolution": "fix this please",
          "Severity": "LOW",
          "References": [
            ""
          ],
          "Status": "FAIL",
          "Layer": {},
          "CauseMetadata": {
            "Provider": "aws",
            "Service": "s3",
            "Code": {
              "Lines": null

**Reply by nikpivkin** [reply]
How do you run Trivy?

**Reply by dave505** [reply]
`My-MacBook-Pro-16-inch-2021- trivy-cloud-scanner % trivy aws --service s3 --config-check ./custom/test.rego --check-namespaces user --format json --output ./test.out`

Where ./custom/test.rego is your rego policy from above

**Reply by nikpivkin** [reply]
Thank you, I'll investigate it

**Comment by micsinyei** [comment]
I would like to add the `custom.id `is not displayed.
If I am testing the [terrafom plan](https://github.com/aquasecurity/trivy-checks/tree/main/examples/terraform-plan)
`trivy config --misconfig-scanners=json --config-check ./asg_capacity.rego --namespaces user ./tfplan.json `
When I use `id `it is not shown, only `avd.id `is displayed (replaced, not using both at once)

**Reply by simar7** [reply]
@micsinyei @dave505  we will be making changes to ID and AVDID shortly and consolidating them into one so this wouldn't be an issue.
